src/llmq/utils.cpp

```diff
@@ -415,7 +456,7 @@ void BuildQuorumSnapshot(const Consensus::LLMQParams& llmqParams, const CDetermi
 {
     quorumSnapshot.activeQuorumMembers.resize(allMns.GetAllMNsCount());
     const CBlockIndex* pWorkBlockIndex = pQuorumBaseBlockIndex->GetAncestor(pQuorumBaseBlockIndex->nHeight - 8);
-    auto modifier = ::SerializeHash(std::make_pair(llmqParams.type, pWorkBlockIndex->GetBlockHash()));
+    const auto modifier = GetHashModifier(llmqParams.type, pWorkBlockIndex);
```

Comment:
@UdjinM6 Agree on the confusing part with `pQuorumBaseBlockIndex` and `pWorkBlockIndex`.
We use height offset only for indexed (rotated) quorums only because:
Members of all indexed quorums are calculated at once at height where `h % dkginterval = 0` (the cycleBlock)
And in each block the next indexed is created.

Reply:
@UdjinM6 Height offset was introduced for indexed (rotation) quorums. With rotation we write quorum snapshots in evoDB: hence we want to protect miners from polluting evoDB (miners try several blocks with different blockhash)